src/dmd/expressionsem.d

```diff
@@ -4113,7 +4113,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         // Type is a "delegate to" or "pointer to" the function literal
         if ((exp.fd.isNested() && exp.fd.tok == TOK.delegate_) || (exp.tok == TOK.reserved && exp.fd.treq && exp.fd.treq.ty == Tdelegate))
         {
-            exp.type = new TypeDelegate(exp.fd.type);
+            exp.type = new TypeDelegate(cast(TypeFunction)exp.fd.type);
```

Comment:
toTypeFunction?

Reply:
Doesn't make a difference, does it ? `fd.type` should really be typed as `TypeFunction`.

Reply:
Nothing wrong with a bit of enforcement then, and an assert will be thrown if someone accidentally assigns something else.

Reply:
Done

Reply:
There are other casts.